odoo/addons/base/models/ir_ui_view.py

```diff
@@ -1067,6 +1079,29 @@ def _postprocess_access_rights(self, tree):
 
         return tree
 
+    def _postprocess_context_dependent(self, tree):
+        """
+        Evaluate the modifiers which depends on the context after retrieving the view from the cache.
+
+        e.g.
+        <field name="date_approve" invisible="context.get('quotation_only', False)"
+
+        For such modifiers, which cannot be cached, the modifier has been stored under it's non-evaluated expression,
+        along with a temporary technical attribute `postprocess_context_modifiers`
+        to tell which modifier should be evaluated after retrieving the view from the cache.
+        e.g.
+        <field
+            name="date_approve"
+            modifiers="{&quot;invisible&quot;: &quot;context.get('quotation_only')&quot;}"
+            postprocess_context_modifiers="invisible"/>
+        """
+        for node in tree.xpath('//*[@postprocess_context_modifiers]'):
+            modifiers = json.loads(node.attrib.pop('modifiers'))
+            for attr in node.attrib.pop('postprocess_context_modifiers').split(','):
+                modifiers[attr] = bool(quick_eval(modifiers[attr], {'context': self.env.context}))
```

Comment:
`quick_eval` seems useless here: by definition we're only leaving attributes which can't be quick-eval'd, since the special cases were already handled before caching.

Reply:
Indeed. I therefore use `safe_eval` directly instead.